pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/MethodReturnsInternalArrayRule.java

```diff
@@ -46,7 +46,7 @@ public Object visit(ASTMethodDeclaration method, Object data) {
             return data;
         }
         List<ASTReturnStatement> returns = method.findDescendantsOfType(ASTReturnStatement.class);
-        ASTTypeDeclaration td = method.getFirstParentOfType(ASTTypeDeclaration.class);
+        ASTClassOrInterfaceDeclaration td = method.getFirstParentOfType(ASTClassOrInterfaceDeclaration.class);
```

Comment:
possibly using `ASTAnyTypeDeclaration` here instead would keep things running and avoid the regression shown in #1845